guide/src/function.md

```diff
@@ -189,3 +189,53 @@ If you have a static function, you can expose it with `#[pyfunction]` and use [`
 [`PyAny::call1`]: https://docs.rs/pyo3/latest/pyo3/struct.PyAny.html#tymethod.call1
 [`PyObject`]: https://docs.rs/pyo3/latest/pyo3/type.PyObject.html
 [`wrap_pyfunction!`]: https://docs.rs/pyo3/latest/pyo3/macro.wrap_pyfunction.html
+
+### Accessing the module of a function
+
+Functions are usually associated with modules, in the C-API, the self parameter in a function call corresponds
+to the module of the function. It is possible to access the module of a `#[pyfunction]` and `#[pyfn]` in the
+function body by passing the `need_module` argument to the attribute:
+
+```rust
+use pyo3::wrap_pyfunction;
+use pyo3::prelude::*;
+
+#[pyfunction(need_module)]
```

Comment:
Bikeshedding: I'd like to propose to call the attribute `pass_module`.

Motivation is that this was already the verb used in #828

Also I've seen this phrasing before e.g. in Python's `click` library: https://click.palletsprojects.com/en/7.x/api/#click.pass_context